### PR TITLE
Refactoring bee species

### PR DESCRIPTION
**Pull request by KorDum**
Removed ExtraBeeSpecies and etc.
Added Extra*Definition like Forestry.

### CODE REVIEW

File: src/main/java/binnie/core/genetics/BreedingSystem.java [hunk 1]
```diff
@@ -145,12 +144,15 @@ public void calculateArrays() {
 				allBranches.add(branch);
 			}
 		}
+
 		if (getSpeciesRoot().getMutations(false) != null) {
 			Set<IMutation> mutations = new LinkedHashSet<>();
 			mutations.addAll(getSpeciesRoot().getMutations(false));
 			if (this == Binnie.Genetics.beeBreedingSystem) {
-				mutations.addAll(ExtraBeeMutation.mutations);
+				// TODO !!! check for needed !!!
```
Comment (mezz) [inline src/main/java/binnie/core/genetics/BreedingSystem.java]: Why is this here?

File: src/main/java/binnie/extrabees/gui/database/ControlProductsItem.java [hunk 2]
```diff
@@ -24,7 +24,7 @@ public ControlProductsItem(ControlList<ControlProductsBox.Product> controlList,
 		}
 
 		item.setItemStack(value.item);
-		float time = (int) (55000.0 / value.chance);
+		float time = (int) (550.0 / value.chance);
```
Comment (mezz) [inline src/main/java/binnie/extrabees/gui/database/ControlProductsItem.java:27]: Why is this changed?

**Review by mezz — CHANGES_REQUESTED** [review]

File: src/main/java/binnie/core/genetics/BreedingSystem.java (hunk 1 above)
Comment (KorDum) [reply src/main/java/binnie/core/genetics/BreedingSystem.java]: Left just in case. I checked, all logics works fine.
Sooth to say, I don't know, why Binnie added this.

File: src/main/java/binnie/extrabees/gui/database/ControlProductsItem.java (hunk 2 above)
Comment (KorDum) [reply src/main/java/binnie/extrabees/gui/database/ControlProductsItem.java:27]: Because production time was displayed 100 times more. 15.2 hours instead 4.2 minutes for example.
I didn't begin to see, why this happened :)

File: src/main/java/binnie/core/genetics/BreedingSystem.java (hunk 1 above)
Comment (mezz) [reply src/main/java/binnie/core/genetics/BreedingSystem.java]: I think to be safe you should leave this alone for now and address it in a different commit. 
Try to leave the logic alone if you are just refactoring and cleaning up code in a commit, that way it's easier to review and look back on what changes were made.

File: src/main/java/binnie/extrabees/gui/database/ControlProductsItem.java (hunk 2 above)
Comment (KorDum) [reply src/main/java/binnie/extrabees/gui/database/ControlProductsItem.java:27]: I checked this bug on old version of code after decompile and port to Forestry 4.2.
The problem is the place to be.

**Review by mezz — CHANGES_REQUESTED** [review]

File: src/main/java/binnie/core/genetics/BreedingSystem.java (hunk 1 above)
Comment (KorDum) [reply src/main/java/binnie/core/genetics/BreedingSystem.java]: You are right. Done.

Comment (p-schneider) [reply src/main/java/binnie/core/genetics/BreedingSystem.java]: The lines were changed/introduced in a commit called "Fixes and stuff for Pre5" (d286440 2015-01-20), but the commit was huge (77 changed files), so I don't understand the reasons for and the effects of this change.

Comment (KorDum) [reply src/main/java/binnie/core/genetics/BreedingSystem.java]: Where can I see old repo for Binnie?

Comment (p-schneider) [reply src/main/java/binnie/core/genetics/BreedingSystem.java]: I looked into [the original project on bitbucket](https://bitbucket.org/binnie567/binnie-mods/commits/d286440deae0c96a352ec239b28e1331bd35108d#Lcore-src/binnie/core/genetics/BreedingSystem.javaF158) where i have read access to the repository, but it is not publicly visible.

Comment (KorDum) [reply src/main/java/binnie/core/genetics/BreedingSystem.java]: Yes, "Access denied". It's a pity.

File: src/main/java/binnie/extrabees/genetics/ExtraBeeDefinition.java [hunk 3]
```diff
@@ -1649,7 +1649,7 @@ protected void setSpeciesProperties(IAlleleBeeSpeciesCustom beeSpecies) {
 
 		@Override
 		protected void registerMutations() {
-			registerMutation(BeeDefinition.ENDED, RELIC, 100)
+			registerMutation(BeeDefinition.ENDED, RELIC, 2)
```
Comment (Elec332) [inline src/main/java/binnie/extrabees/genetics/ExtraBeeDefinition.java]: Why would you change this?

**Review by Elec332 — CHANGES_REQUESTED** [review]

File: src/main/java/binnie/extrabees/genetics/ExtraBeeDefinition.java (hunk 3 above)
Comment (KorDum) [reply src/main/java/binnie/extrabees/genetics/ExtraBeeDefinition.java]: Because chance of this mutation is 2 in original Binnie: https://github.com/ForestryMC/Binnie/blob/master-MC1.7.10/src/main/java/binnie/extrabees/genetics/ExtraBeeMutation.java#L203
I did 100% for the test condition and accidentally made a commit with this.
